velox/connectors/hive/iceberg/tests/IcebergDataSinkTest.cpp

```diff
@@ -0,0 +1,388 @@
+#include "velox/exec/tests/utils/HiveConnectorTestBase.h"
+
+#include <folly/init/Init.h>
+#include "velox/common/base/Fs.h"
+#include "velox/dwio/common/BufferedInput.h"
+#include "velox/dwio/dwrf/reader/DwrfReader.h"
+#include "velox/dwio/dwrf/writer/Writer.h"
+
+#ifdef VELOX_ENABLE_PARQUET
+#include "velox/dwio/parquet/RegisterParquetWriter.h"
+#include "velox/dwio/parquet/reader/ParquetReader.h"
+#endif
+
+#include "velox/exec/tests/utils/PlanBuilder.h"
+#include "velox/exec/tests/utils/TempDirectoryPath.h"
+
+namespace facebook::velox::connector::hive::iceberg {
+namespace {
+
+using namespace facebook::velox::common;
+using namespace facebook::velox::exec::test;
+using namespace facebook::velox::common::testutil;
+
+class IcebergDataSinkTest : public exec::test::HiveConnectorTestBase {
+ protected:
+  void SetUp() override {
+    HiveConnectorTestBase::SetUp();
+#ifdef VELOX_ENABLE_PARQUET
+    parquet::registerParquetReaderFactory();
+    parquet::registerParquetWriterFactory();
+#endif
+    Type::registerSerDe();
+
+    rowType_ =
+        ROW({"c0", "c1", "c2", "c3", "c4", "c5", "c6"},
+            {BIGINT(),
+             INTEGER(),
+             SMALLINT(),
+             REAL(),
+             DOUBLE(),
+             VARCHAR(),
+             BOOLEAN()});
+
+    setupMemoryPools();
+  }
+
+  void TearDown() override {
+    connectorQueryCtx_.reset();
+    connectorPool_.reset();
+    opPool_.reset();
+    root_.reset();
+    HiveConnectorTestBase::TearDown();
+  }
+
+  std::vector<RowVectorPtr> createVectors(int vectorSize, int numVectors) {
+    VectorFuzzer::Options options;
+    options.vectorSize = vectorSize;
+    VectorFuzzer fuzzer(options, pool());
+    std::vector<RowVectorPtr> vectors;
+    for (int i = 0; i < numVectors; ++i) {
+      vectors.push_back(fuzzer.fuzzInputRow(rowType_));
+    }
+    return vectors;
+  }
+
+  void setupMemoryPools() {
+    connectorQueryCtx_.reset();
+    connectorPool_.reset();
+    opPool_.reset();
+    root_.reset();
+
+    root_ = memory::memoryManager()->addRootPool(
+        "IcebergDataSinkTest", 1L << 30, exec::MemoryReclaimer::create());
+    opPool_ = root_->addLeafChild("operator");
+    connectorPool_ =
+        root_->addAggregateChild("connector", exec::MemoryReclaimer::create());
+
+    connectorQueryCtx_ = std::make_unique<connector::ConnectorQueryCtx>(
+        opPool_.get(),
+        connectorPool_.get(),
+        connectorSessionProperties_.get(),
+        nullptr,
+        common::PrefixSortConfig(),
+        nullptr,
+        nullptr,
+        "query.IcebergDataSinkTest",
+        "task.IcebergDataSinkTest",
+        "planNodeId.IcebergDataSinkTest",
+        0,
+        "");
+  }
+
+  std::shared_ptr<connector::hive::iceberg::IcebergInsertTableHandle>
+  createIcebergInsertTableHandle(
+      const RowTypePtr& outputRowType,
+      const std::string& outputDirectoryPath,
+      dwio::common::FileFormat fileFormat = dwio::common::FileFormat::DWRF,
```

Comment:
Why use DWRF here?

Reply:
I tried to maintain consistency with other iceberg tests by adding tests for both DWRF and Parquet formats.  Please let me know if any changes are needed.